File: extract.js
const fs = require('fs');
const PNG = require('pngjs').PNG;

function extractColors(imagePath) {
    fs.createReadStream(imagePath)
    .pipe(new PNG())
    .on('parsed', function() {
        const colorCounts = {};
        for (let y = 0; y < this.height; y++) {
            for (let x = 0; x < this.width; x++) {
                const idx = (this.width * y + x) << 2;
                const r = this.data[idx];
                const g = this.data[idx+1];
                const b = this.data[idx+2];
                const a = this.data[idx+3];
                
                if (a > 10) { // ignore very transparent pixels
                    const hex = '#' + 
                        r.toString(16).padStart(2, '0') + 
                        g.toString(16).padStart(2, '0') + 
                        b.toString(16).padStart(2, '0');
                    colorCounts[hex] = (colorCounts[hex] || 0) + 1;
                }
            }
        }
        
        const sortedColors = Object.entries(colorCounts)
            .sort((a, b) => b[1] - a[1])
            .slice(0, 5);
            
        console.log(`Top colors for ${imagePath}:`);
        sortedColors.forEach(([hex, count]) => {
            console.log(`${hex} - count: ${count}`);
        });
    });
}

extractColors('docs/temp/Layout/Logo Transparente/LOGO PADRÃO-1.png');
extractColors('docs/temp/Layout/Logo Transparente/LOGO BR-1.png');

File: .aiox-core/infrastructure/scripts/design-tools/extract.py
import sys
try:
    from PIL import Image
except ImportError:
    print("PIL not installed")
    sys.exit(0)

def extract_colors(image_path):
    img = Image.open(image_path).convert("RGBA")
    colors = img.getcolors(maxcolors=1000000)
    # Sort by count (descending)
    colors.sort(key=lambda x: x[0], reverse=True)
    
    print(f"Top 10 colors for {image_path}:")
    for count, rgba in colors[:10]:
        if rgba[3] > 0: # ignore transparent
            hex_color = "#{:02x}{:02x}{:02x}".format(rgba[0], rgba[1], rgba[2])
            print(f"{hex_color} - count: {count}")

try:
    extract_colors(r"c:\Users\Gabriel Cristofolini\Documents\SOLUCOESSISTEMAS\tech-arauz\docs\temp\Layout\Logo Transparente\LOGO PADRÃO-1.png")
except Exception as e:
    print(f"Error: {e}")
